### ISSUE TITLE
MPV font handling

### ISSUE THREAD

**Issue by OmrSi** [SubtitleEdit/subtitleedit#3282]
I think there might be a problem here.
When MPV handles font, it should be handled as ass/ssa and therefore the font size should be determined by the size in the ass/ssa window, not the size in the Video window.
This is causing problems like this:
I tried font Mitra LT Bold with size 30 in the video window, which is the max, and this is what I got:
![screenshot 72](https://user-images.githubusercontent.com/20923700/50743905-cf266280-1225-11e9-8184-10e8a8e09af6.png)

I can't read it, and since the font size in ass changes based on the video's resolution, sometimes it looks good and sometimes it looks too small.
That's why I think that the font size when using MPV should be based on the font size in the ass/ssa window.

**Comment by niksedk** [comment]
I've given it a try... how is this: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.8/SubtitleEditBeta.zip ?

**Comment by OmrSi** [comment]
The size is better now for small fonts but it still relays on the size in Video window not in ass/ssa.
You can close this if this is intended.

**Comment by OmrSi** [comment]
@niksedk  The size is better now for small fonts but it still relays on the size in Video window not in ass/ssa.
You can close this if this is intended.
What I meant was that MPV handles the preview as an ass file, so even the font size should be determined by the ass window, while the size in the video window is for the other video players as ass can handle larger font sizes...

**Comment by OmrSi** [comment]
@niksedk In case you missed it, I edited the above commit and added some explanation...

**Comment by niksedk** [comment]
I'm not really sure what you mean.... but beta updated: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.8/SubtitleEditBeta.zip

**Comment by OmrSi** [comment]
I don't know how to explain it more. 😅
All I'm saying is that when using MPV as the video player and letting it handle font preview, the font size should be taken from here:
![screenshot 84](https://user-images.githubusercontent.com/20923700/51079442-d50dbf00-16cf-11e9-83f6-fb507cdb13fc.png)
Because it handles it as an ass file, and the one in Video should be for the font size in other players.

By the way, I seem to be unable to undo sometimes after pasting in the latest beta:
![screenshot 85](https://user-images.githubusercontent.com/20923700/51079494-9c221a00-16d0-11e9-9619-c50acf4f80c7.png)

**Comment by niksedk** [comment]
Ah, the font size is taken from Settings -> Video player -> Subtitle preview font size:
![image](https://user-images.githubusercontent.com/1261331/51082903-656bf400-1710-11e9-93e0-d2a7a5e71913.png)
(the rest of the settings are from the default ssa/ass style)

**Comment by OmrSi** [comment]
Yeah, that's what I meant in the first place.
As MPV handles it as an ass file, the Subtitle preview font size should be determined from the default ssa/ass style.
The reason I'm saying this is that some fonts are too small for the size in "Video player" window.
When you use 30 for Directshow or MPC or VLC, the font will look large because it's one font and its size is fixed.
![screenshot 86](https://user-images.githubusercontent.com/20923700/51086209-75092e00-174c-11e9-9f03-030a0a4da1d0.png)

But MPV is not the same, different fonts have different sizes so 30 might not be big enough:
![screenshot 72](https://user-images.githubusercontent.com/20923700/50743905-cf266280-1225-11e9-8184-10e8a8e09af6.png)

The default ssa/ass style for MPV is better because it can take larger font sizes than just 30.
bottom line, Subtitle preview font size should be taken from the default ssa/ass style for MPV.

If it's too complicated or you'd prefer it to all be from one place, you should just close this. 😄

**Comment by OmrSi** [comment]
@niksedk I think you should undo this. It's all messed up now.
It's become worse with srt files in fullscreen and in ass file without fullscreen:
![screenshot 100](https://user-images.githubusercontent.com/20923700/51192867-2eeccf80-18f0-11e9-8529-5380d0af9874.png)
![screenshot 101](https://user-images.githubusercontent.com/20923700/51192951-593e8d00-18f0-11e9-8d6b-8aadf2f07c2e.png)

Font size handling was good in MPV, I was just saying that the size should be taken from the default Style with MPV.

**Comment by OmrSi** [comment]
@niksedk Could you undo the changes? I need to QC a file and the font is not good with ass file or srt in fullscreen and \pos tags aren't working well now and I don't have an old beta. :\
![screenshot 101](https://user-images.githubusercontent.com/20923700/51349257-7f5d5c00-1aad-11e9-9e88-e306bb4c8227.png)

I only wanted the font size with MPV to be taken from the Default style because some fonts are too small for the size in "Video player" window with MPV but "Video player" preview size is good for other players because they have one non-changing size.

**Comment by OmrSi** [comment]
@niksedk Could you undo the changes? I need to QC a file and the font is not good with ASS file or SRT in fullscreen and \pos tags aren't working well now and I don't have an old beta. :\
![screenshot 100](https://user-images.githubusercontent.com/20923700/51349370-dc591200-1aad-11e9-93cd-2f1e1236b251.png)

I only wanted the font size with MPV to be taken from the Default style because some fonts are too small for the size in "Video player" window with MPV but "Video player" preview size is good for other players because they have one non-changing size.
![untitled](https://user-images.githubusercontent.com/20923700/51350569-2d1e3a00-1ab1-11e9-84da-ace798ccde6f.png)

**Comment by niksedk** [comment]
How is latest beta?
https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.8/SubtitleEditBeta.zip

I've tried to keep the font size for non-ass subtitles... font size should from mpv and non-mpv should be somewhat comparable.
ASS subtitles are back to old functionality.

**Comment by OmrSi** [comment]
@niksedk It's better the way it was, look:
![screenshot 106](https://user-images.githubusercontent.com/20923700/51398976-83d95180-1b4d-11e9-8c18-25268e9971da.jpg)

Making the font bigger inside SE isn't good if it's not the same with external players, it will cause a problem with {\fsXX} and {\pos} and <font size=> tags when played in external players, it won't be compatible with it.
That's why I only suggested taking the size from the default with MPV, but if that's not how you want it, you can close this. :D
